sklearn/feature_extraction/dict_vectorizer.py

```diff
@@ -95,7 +96,30 @@ def __init__(self, dtype=np.float64, separator="=", sparse=True,
         self.sparse = sparse
         self.sort = sort
 
-    def fit(self, X, y=None):
+    @staticmethod
+    def _validate_types(X):
+        """Ensure that features have consistent types and raise ValueError if
+        numbers and strings are mixed.
+
+        Parameters
+        ----------
+        X : Mapping or iterable over Mappings
+            Dict(s) or Mapping(s) from feature names (arbitrary Python
+            objects) to feature values (strings or convertible to dtype).
+
+        Raises
+        ------
+        ValueError
+            If any feature in ``X`` has mixed (both numeric and text) types.
+        """
+        numeric_X = {}
+        for f in X:
+            for k, v in six.iteritems(f):
+                if k in numeric_X and numeric_X[k] != isinstance(v, Number):
+                    raise ValueError('Multiple types for feature "%s".' % (k))
+                numeric_X[k] = isinstance(v, Number)
+
+    def fit(self, X, y=None, validate_types=True):
```

Comment:
The `validate_types` flag is not needed. I feel we should always raise error if there are incompatible types. So could you remove this flag and make it the default behaviour?